google/oauth2/_client.py

```diff
@@ -505,3 +509,81 @@ def refresh_grant(
         request, token_uri, body, can_retry=can_retry
     )
     return _handle_refresh_grant_response(response_data, refresh_token)
+
+
+def lookup_trust_boundary(request, universe, service_account, access_token):
+    """ Implements the global lookup of service account trust boundary
+
+    Args:
+        request (google.auth.transport.Request): A callable used to make
+            HTTP requests.
+
+        universe (str): Universe domain. (GDU/TPC universe)
+
+        service_account (str): The service account to lookup.
+
+        access_token (str): A valid access_token used as a basic auth.
+
+    Returns:
+        Mapping[str,list|str, str]: A response would be a dictionary contains
+            the "locations" as a list of allowed locations in string and
+            "encoded_locations" as a hex string.
+
+            e.g:
+            {
+                "locations": [
+                    "us-central1", "us-east1", "europe-west1", "asia-east1"
+                ],
+                "encoded_locations": "0xA30"
+            }
+
+            In case of a universe global lookup not launched yet, a default
+            trust boundary of "all" will be returned.
+
+            {
+                "locations": [],
+                "": "0x0"
+            }
+
+    Raises:
+        exceptions.TransportError: If the request to lookup endpoint fails.
+        exceptions.RefreshError: If the query response not 200 or 404.
+        exceptions.MalformedError: If the response not in valid format.
+    """
+    url = _TRUST_BOUNDARY_LOOKUP_ENDPOINT.format(universe, service_account)
```

Comment:
Are the trust boundary endpoints going to be active for PSC environments? If so, this needs to be overriddable, if not, we should confirm that in a PSC environment, this flow is ignored. I think it would probably return not found, but not 100% sure, and if it doesn't this change would break all PSC users.

Reply:
If we don't launch in any environment, we will go for a flow for `404` which fall backs to the current behavior

Reply:
Actually, thinking about it more, I'm worried about this breaking the flow in general. The trust boundary backend isn't actually live yet right? And even when it does go live it isn't actually required for BYOID? If the trust boundary service is broken (say returning a 503) this logic would break auth for everyone. I would think we would want to ignore it in that case.

Reply:
The `refresh` method of external_account don't do this part. We just have the hard coded "enable all" values in the credential objects and apply it to requests when "apply" it.

Reply:
@aeitzman Did some modification only processing the potential errors under `200` return. For others we fall back to the old logic to give a default trust boundary.

Reply:
Sure, sounds good. Let's wait to finalize on this until we get a good answer from the backend team though.